x/mint/types/params.go

```diff
@@ -107,32 +101,19 @@ func validateGenesisEpochProvisions(i interface{}) error {
 	return nil
 }
 
-func validateReductionPeriodInEpochs(i interface{}) error {
-	v, ok := i.(int64)
-	if !ok {
-		return fmt.Errorf("invalid parameter type: %T", i)
-	}
-
-	if v <= 0 {
-		return fmt.Errorf("max validators must be positive: %d", v)
-	}
-
-	return nil
-}
-
-func validateReductionFactor(i interface{}) error {
-	v, ok := i.(sdk.Dec)
+func validateTokenReleaseSchedule(i interface{}) error {
+	tokenReleaseSchedule, ok := i.([]ScheduledTokenRelease)
```

Comment:
can we also check that date is in the right "yyyy-mm-dd" format, and other basic things like:
- mm, dd > 0
- yyyy > 1970
- mm < 12
- dd <= 31
- dd logic:
- - if mm == 1,3,5,7,8,10,12, dd == 31
- - if mm == 4,6,9,11, dd ==30
- - if mm == 2 and yy % 4 != 0, dd == 28
- - if mm == 2 and yy % $ == 0, dd == 29
Re. leap year, seems like the logic is a bit more complicated:
```
Years that are divisible by 100 (century years such as 1900 or 2000) cannot be leap years unless they are also divisible by 400. (For this reason, the years 1700, 1800, and 1900 were not leap years, but the years 1600 and 2000 were.)
```
but think we can ignore this, maybe just make a note

Reply:
Hmm, I was hoping that the DateTime parser should be catching almost all the cases you mentioned above 

`TokenReleaseDateFormat == yyyy-mm-dd` here
```
		_, err := time.Parse(TokenReleaseDateFormat, scheduledTokenRelease.GetDate())
```

Reply:
e.g 
<img width="807" alt="image" src="https://user-images.githubusercontent.com/18161326/189027831-528248bf-1c71-43a8-9655-ecd8437a9f01.png">

Reply:
one thing it's not checking is if the year is > 1970 but if it is we won't be hitting that condition anyways